### PR TITLE
Drop py2 support

### PR DESCRIPTION
**Pull request by ki4070ma**
For https://github.com/appium/python-client/issues/468

[Notes]
* Merge after April 2020
* Need check branch strategy
   * `master` branch is for v1.0?
   * Or create `v1.0` branch as feature branch?
   * create `v0`(need better naming) as maintenance branch

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,12 +2,11 @@ language: python
 dist: xenial
 
 python:
-  - "2.7"
   - "3.4"
```
Comment (ki4070ma) [inline .travis.yml]: Delete 3.4, 3.5, 3.6

File: setup.py [hunk 2]
```diff
@@ -39,11 +39,10 @@
     classifiers=[
         'Development Status :: 4 - Beta',
         'Programming Language :: Python',
-        'Programming Language :: Python :: 2.7',
-        'Programming Language :: Python :: 3.4',
         'Programming Language :: Python :: 3.5',
```
Comment (ki4070ma) [inline setup.py]: Delete 3.5, 3.6

File: tox.ini [hunk 3]
```diff
@@ -1,12 +1,11 @@
 [tox]
 skipsdist = True
 envlist =
-    py27,
     py34,
```
Comment (ki4070ma) [inline tox.ini]: Delete py34, py35, py36

**Comment by ki4070ma** [comment]
@mykola-mokhnach @KazuCocoa 
How will we handle branches for future? 

e.g.
* `master` branch is for v1.0?
* create `v0` (need better naming) as maintenance branch

**Comment by ki4070ma** [comment]
Reopened since it's my mistake.

**Comment by KazuCocoa** [comment]
We should wait for https://www.python.org/psf/press-release/pr20191220/, I believe.

Then, `master` can be `v1` and we can also cut `v0` for Python 2.7 to Python 3.6.
We will develop/maintain only master (v1). But if someone is interested in porting `master`'s changes to v0 branch, they can create PRs.
What do you think?

**Comment by ki4070ma** [comment]
@KazuCocoa 

Thanks for your comments.

> We should wait for https://www.python.org/psf/press-release/pr20191220/, I believe.

Thanks, I didn't know that.
We can wait April 2020, but it's not mandatory to wait in my opinion.
However let's wait April 2020 since there is no strong reason to rush it.

* 1st Jan 2020: Stop support py27
* April 2020: The final release for py27

> We will develop/maintain only master (v1).
ok, I agree.

> But if someone is interested in porting master's changes to v0 branch, they can create PRs.
Good idea.

**Comment by mykola-mokhnach** [comment]
First, lets create `python3` branch and work on basic py3 compatibility features. Then we merge this branch into master when python 2 is finally EOLed. Before the `python3` branch is merged to master we rebase it and create `python2` branch, where we keep the most recent `master` state before completely dropping Python 2 support. At the end we still only maintain `master` branch with Python 3 support, publish the new major driver version and cherry pick minor fixes to `python2` branch if needed. The `python2` branch will be the base for the previous major driver version and could be used to provide critical patches to it.

**Comment by mykola-mokhnach** [comment]
Action sequence in more readable representation:

1. Create `python3` branch where we drop Py2 support
2. Rebase `python3` branch with `master` before it is merged
3. Create `python2` branch from `master` before `pytnon3` is merged
4. Merge `pytnon3` branch to `master`
5. Publish new major client version
6. Cherry pick minor fixes to `python2` branch if necessary

**Comment by ki4070ma** [comment]
@mykola-mokhnach 
Thanks for your detailed explanation.
As your suggestion, it's reasonable to start python3 adaptation in early.

Now I've changed this PR target to `python3` branch.

**Pull request merged**